### DISCUSSION TITLE
Why does the git status change to modified after open by vscode?

### DISCUSSION THREAD

**Discussion by AllenMH** [microsoft/vscode-cpptools#10184] [Q&A]
when installs and enables cpptools in vscode on windows, open the C++ code directory through vscode, and then open a file, such as: a.h, sometimes the git status of the file will change to modified, and compare the file content by binary, but we have not made any changes to the file ,there is no difference in one byte including carriage return, space, tab key, etc.
After disabling the plugin there is no such problem.
Using the Procmon.exe tool to monitor, it is found that code.exe does open and write files through functions such as CreateFileW, WriteFile, and FlushFileBuffers. The content of the file has not changed after writing (presumably the same content is written), but the file time is different change.
Try to write a test program, call CreateFileW, ReadFile, WriteFile, FlushFileBuffers functions to simulate writing the same content. After execution, the file time changed, and the git status does not change to modified.
which is in line with expectations.
Why does cpptools open files with this change?

**Comment by sean-mcmanus** [comment]
We haven't seen this behavior. Do you have any more repro details? We're not aware of any feature that updates a file, unless if you have some sort of auto-formatting enabled.
